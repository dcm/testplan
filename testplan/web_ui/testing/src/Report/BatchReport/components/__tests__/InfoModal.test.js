/** @jest-environment jsdom */
// @ts-nocheck
import React from 'react';
import { shallow } from 'enzyme';
import { StyleSheetTestUtils } from 'aphrodite';
import { InfoModal } from '../';
import useReportState from '../../hooks/useReportState';

jest.mock('../../hooks/useReportState');

describe('InfoModal', () => {

  beforeAll(() => {
    useReportState.mockName('useReportState');
  });

  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  it('renders as expected when modal is showing', () => {
    useReportState.mockReturnValue([ true, jest.fn() ]);
    // we're using `shallow` because we don't want the results of these tests
    // to depend upon the child component <InfoTable/>
    const wrapper = shallow(<InfoModal/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders as expected when modal is NOT showing', () => {
    useReportState.mockReturnValue([ false, jest.fn() ]);
    const wrapper = shallow(<InfoModal/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('grabs correct slices from useReportState', () => {
    const expectedHookArgs = [
      'app.reports.batch.isShowInfoModal',
      'setAppBatchReportShowInfoModal',
    ];
    useReportState.mockReturnValue([
      false, jest.fn().mockName(expectedHookArgs[1])
    ]);
    shallow(<InfoModal/>);
    expect(useReportState).toHaveBeenCalledTimes(1);
    expect(useReportState).toHaveBeenLastCalledWith(...expectedHookArgs);
  });

  it('"Close" button calls `setAppBatchReportShowInfoModal`', () => {
    const expectedHookArgs = [
      'app.reports.batch.isShowInfoModal',
      'setAppBatchReportShowInfoModal',
    ];
    const setShowInfoModal = jest.fn().mockName(expectedHookArgs[1]);
    const isShowInfoModal = true;
    useReportState.mockReturnValue([ isShowInfoModal, setShowInfoModal ]);
    const wrapper = shallow(<InfoModal/>);
    expect(useReportState).toHaveBeenCalledTimes(1);
    expect(useReportState).toHaveBeenLastCalledWith(...expectedHookArgs);
    wrapper.find('Button').simulate('click');
    expect(setShowInfoModal).toHaveBeenCalledTimes(1);
    expect(setShowInfoModal).toHaveBeenLastCalledWith(!isShowInfoModal);
  });

  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
    jest.resetAllMocks();
  });

});
