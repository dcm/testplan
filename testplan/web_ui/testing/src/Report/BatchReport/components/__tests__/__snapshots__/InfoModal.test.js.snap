// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoModal renders as expected when modal is NOT showing 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  centered={false}
  className="infoModal"
  fade={true}
  isOpen={false}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  role="dialog"
  size="lg"
  toggle={[Function]}
  zIndex={1050}
>
  <ModalHeader
    closeAriaLabel="Close"
    tag="h5"
    toggle={[Function]}
    wrapTag="div"
  >
    Information
  </ModalHeader>
  <ModalBody
    tag="div"
  >
    <InfoTable />
  </ModalBody>
  <ModalFooter
    tag="div"
  >
    <Button
      color="light"
      onClick={[Function]}
      tag="button"
    >
      Close
    </Button>
  </ModalFooter>
</Modal>
`;

exports[`InfoModal renders as expected when modal is showing 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  centered={false}
  className="infoModal"
  fade={true}
  isOpen={true}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  role="dialog"
  size="lg"
  toggle={[Function]}
  zIndex={1050}
>
  <ModalHeader
    closeAriaLabel="Close"
    tag="h5"
    toggle={[Function]}
    wrapTag="div"
  >
    Information
  </ModalHeader>
  <ModalBody
    tag="div"
  >
    <InfoTable />
  </ModalBody>
  <ModalFooter
    tag="div"
  >
    <Button
      color="light"
      onClick={[Function]}
      tag="button"
    >
      Close
    </Button>
  </ModalFooter>
</Modal>
`;
